### ISSUE TITLE
Calling 'this' returns wrong values when used in expression trees.

### ISSUE THREAD

**Issue by ItsDeltin** [ItsDeltin/Overwatch-Script-To-Workshop#135]
```
class MyClass
{
    AnotherClass anotherClass;

    public void Test()
    {
        // The 'this' here will return 'anotherClass' rather than the current MyClass instance.
        anotherClass.DoSomething(this);
    }
}
```

**Comment by ItsDeltin** [comment]
Workaround is to store `this`:
```
define realThis! = this;
anotherClass.DoSomething(realThis);
```

**Comment by ItsDeltin** [comment]
Fixed in [v1.3](https://github.com/ItsDeltin/Overwatch-Script-To-Workshop/releases/tag/v1.3).
